v3/src/components/graph/adornments/lsrl/lsrl-adornment-model.ts

```diff
@@ -0,0 +1,157 @@
+    return { lower: tYHat - tAddend, upper: tYHat + tAddend }
+  }
+}))
+.views(self => ({
+  confidenceBandsPoints(
+    min: number, max: number, xCellCount: number, yCellCount: number, gap: number, caseValues: Point[],
+    xScale: ScaleNumericBaseType, yScale: ScaleNumericBaseType, cellKey: Record<string, string>, lineIndex: number
+  ) {
+    const upperPoints: any = []
+    const lowerPoints: any = []
+    for (let pixelX = min; pixelX <= max; pixelX += gap) {
+      const tX = xScale.invert(pixelX * xCellCount)
+      const tYValues = self.confidenceValues(tX, caseValues, cellKey, lineIndex)
+      if (!tYValues) continue
+      upperPoints.push({ x: pixelX, y: yScale(tYValues.upper) / yCellCount })
+      lowerPoints.unshift({ x: pixelX, y: yScale(tYValues.lower) / yCellCount })
+    }
+    return { upperPoints, lowerPoints }
+  }
+}))
+.actions(self => ({
+  updateLines(line: {intercept: number, rSquared: number, slope: number, sdResiduals: number}, key="") {
+    const existingLines = self.lines.get(key)
+    const newLines = existingLines ? [...existingLines] : []
+    newLines.push(LSRLInstance.create(line))
+    self.lines.set(key, newLines)
+  },
+  setShowConfidenceBands(showConfidenceBands: boolean) {
+    self.showConfidenceBands = showConfidenceBands
+  },
+  computeValues(
+    xAttrId: string, yAttrId: string, cellKey: Record<string, string>,
+    dataConfig: IGraphDataConfigurationModel, key="", isInterceptLocked=false,
+    cat?: string
+  ) {
+    const caseValues = self.getCaseValues(xAttrId, yAttrId, cellKey, dataConfig, cat)
+    const { intercept, rSquared, slope, sdResiduals } =
+      leastSquaresLinearRegression(caseValues, isInterceptLocked)
+    return { intercept, rSquared, slope, sdResiduals }  
+  }
+}))
+.actions(self => ({
+  updateCategories(options: IUpdateCategoriesOptions) {
+    const { xAttrId, yAttrId, topCats, rightCats, dataConfig } = options
+    if (!dataConfig) return
+    self.lines.clear()
+    const columnCount = topCats?.length || 1
+    const rowCount = rightCats?.length || 1
+    const totalCount = rowCount * columnCount
+    const legendCats = dataConfig?.categoryArrayForAttrRole("legend")
+    for (let i = 0; i < totalCount; ++i) {
+      const cellKey = self.cellKey(options, i)
+      const instanceKey = self.instanceKey(cellKey)
+      for (let j = 0; j < legendCats.length; ++j) {
+        // TODO: Once the Intercept Locked feature is implemented, we will need to pass in something like 
+        // isInterceptLocked instead of false in the call to self.computeValues.
+        const { intercept, rSquared, slope, sdResiduals } = self.computeValues(
+          xAttrId, yAttrId, cellKey, dataConfig, instanceKey, false, legendCats[j]
+        )
+        if (
+          (intercept === null && intercept !== 0) ||
+          (rSquared === null && rSquared !== 0) ||
+          (slope === null && slope !== 0) ||
+          (sdResiduals === null && sdResiduals !== 0)
```

Comment:
A value that `=== null` is by definition `!== 0`.

Reply:
Thanks. I still get confused sometimes by `null`, `0`, and `undefined` and which is considered equal to which. I need to get that squared away in my head.